packages/ckeditor5-image/src/imageinsert/imageinsertviaurlui.ts

```diff
@@ -147,10 +123,48 @@ export default class ImageInsertViaUrlUI extends Plugin {
 	}
 
 	/**
-	 * Closes the dropdown.
+	 *
 	 */
-	private _closePanel(): void {
-		this.editor.editing.view.focus();
-		this._imageInsertUI.dropdownView!.isOpen = false;
+	private _showModal() {
+		const editor = this.editor;
+		const locale = editor.locale;
+		const t = locale.t;
+		const dialog = editor.plugins.get( 'Dialog' );
+
+		const form = this._createInsertUrlView();
+
+		function handleSave( form: ImageInsertUrlView ) {
+			const replaceImageSourceCommand: ReplaceImageSourceCommand = editor.commands.get( 'replaceImageSource' )!;
+
+			if ( replaceImageSourceCommand.isEnabled ) {
```

Comment:
I know you copy-pasted this part but it demands some comment because it is confusing.
```suggestion
			// If an image element is currently selected, we want to replace its source attribute (instead of inserting a new image).
			// We detect if an image is selected by checking `replaceImageSource` command state.
			if ( replaceImageSourceCommand.isEnabled ) {
```